android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -68,6 +71,18 @@ class TranslationsMiddleware(
                     -> Unit
                 }
             }
+
+            is TranslationsAction.FetchTranslationDownloadSizeAction -> {
+                scope.launch {
+                    requestTranslationSize(
+                        context = context,
+                        tabId = action.tabId,
```

Comment:
wondering if the download size is correlated to a tab or if this should also be part of the app state TranslationsBrowserState

Reply:
Thanks for taking a look!

Great thing to mention, I was wondering that too. 

My reasoning is I think it makes sense being on the tab because the expected use case is calling this action whenever a translation is about to be requested, so it'll always follow a translate call when the device is in low data mode, so the data need is tab specific in that sense.

For example:
* User presses translate using from `fr` to `en` on tab 1 on low data mode
* Action will follow to request the size to show the user in the prompt for tab 1 and store it
* Prompt will show `fr` to `en` info
* User navigates to tab 2 without clearing prompt
* On tab 2 user presses translate using `es` to `en` on tab 2 in low data mode
* Action will follow to request the size to show the user in the prompt for tab 2 and store it
* Prompt will show `es` to `en` info

If the result was on the global object, then switching back to tab 1 would be required to re-request the data, since the global data would be storing `es` to `en` via tab 2.  So, I think keeping it on the tab might be more conceptually clear for the expected use case. I think we might also have to be more careful about potentially losing the requested translation state.

However, I think it could be valid on the global object and we would just need to expect to re-request and be more careful about managing state.  LMKWYT!

Reply:
Gotcha, thanks for explaining the ui flow